src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs

```diff
@@ -88,9 +88,12 @@ public void GetObjectData(SerializationInfo info, StreamingContext context)
 
             if (m_localValues == null || AsyncLocalValueMap.IsEmpty(m_localValues))
             {
-                return isFlowSuppressed ?
-                    DefaultFlowSuppressed :
-                    null; // implies the default context
+                if (!isFlowSuppressed)
+                    return null; // implies the default context
+
+                s_DefaultFlowSuppressed ??= new ExecutionContext(AsyncLocalValueMap.Empty, Array.Empty<IAsyncLocal>(), isFlowSuppressed: true);
```

Comment:
Array.Empty use here is not profitable. This can just use `new IAsyncLocal[0]`. The whole thing is singleton and the chance that there will be more than use of `Array.Empty<IAsyncLocal>` in typical program is nil.